lib/tasks/convert_embeddables.rake

```diff
@@ -0,0 +1,326 @@
+    backgroundImageUrl: item.embeddable.bg_url,
+    backgroundSource: background_source,
+    imageFit: "shrinkBackgroundToCanvas",
+    imagePosition: "center",
+    answerPrompt: item.embeddable.drawing_prompt,
+    prompt: item.embeddable.prompt,
+    hint: item.embeddable.hint
+  }
+
+  return name, authored_state.to_json
+end
+
+def convert_video_player(item, library_interactive)
+  # get the MP4 version if present, otherwise take the first URL in the source array
+  video_source = ""
+  item.embeddable.sources.each do |source|
+    if source.format == "video/mp4"
+      video_source = source.url
+    end
+  end
+  if video_source == ""
+    video_source = item.embeddable.sources[0].url
+  end
+
+  authored_state = {
+    version: 1,
+    questionType: "iframe_interactive",
+    videoUrl: video_source,
+    caption: item.embeddable.caption,
+    credit: item.embeddable.credit,
+    creditLinkDisplayText: "",
+    fixedAspectRatio: "",
+    poster: item.embeddable.poster_url
+  }
+
+  return authored_state.to_json
+end
+
+def new_properties(item, library_interactive)
+  new_properties = {
+    library_interactive_id: library_interactive.id,
+    url_fragment: nil,
+    inherit_aspect_ratio_method:	true,
+    custom_aspect_ratio_method: nil,
+    inherit_native_width: true,
+    custom_native_width: library_interactive.native_width,
+    inherit_native_height: true,
+    custom_native_height: library_interactive.native_height,
+    inherit_click_to_play: true,
+    custom_click_to_play: false,
+    inherit_full_window: true,
+    custom_full_window: false,
+    inherit_click_to_play_prompt: true,
+    custom_click_to_play_prompt: nil,
+    inherit_image_url: true,
+    custom_image_url: nil
+  }
+
+  return new_properties
+end
+
+def save_new_item(item, legacy_ref_id, name, authored_state, new_properties)
+  new_embeddable = ManagedInteractive.new(
+    name: name,
+    authored_state: authored_state,
+    legacy_ref_id: legacy_ref_id
+  )
+  new_properties.each do |key, value|
+    new_embeddable[key] = value
+  end
+  new_embeddable.save!
+  # We probably DON'T want to create a new page item for the real migration. This is just for testing.
+  # new_page_item = PageItem.new(position: item.position, section: item.section, interactive_page: item.interactive_page)
+  # new_page_item.embeddable = new_embeddable
+  # new_page_item.save!
+end
+
+def convert()
+  # Get a list of official Library Interactives so we know which ones to convert to.
+  library_interactives = {}
+  LibraryInteractive.where(:official => true).each do |library_interactive|
+    @convertable_embeddable_types.each do |question_type|
+      if library_interactive.name == question_type
+        library_interactives[question_type] = library_interactive
+      end
+    end
+  end
+
+  if library_interactives.length == 0
+    raise Exception.new "No official library interactives found."
+  end
+
+  # activities_to_update = LightweightActivity.where(:runtime => "LARA")
```

Comment:
We might get a small performance benefit by iterating over the PageItems with a specific embeddable type. This way a smaller bit of code is going run again and again during the loop, so the ruby interpreter might optimize it better.  But I think it is best to worry about that later.